pkg/resource/nodegroup/delta.go

```diff
@@ -189,9 +189,29 @@ func newResourceDelta(
 	if !ackcompare.MapStringStringEqual(ToACKTags(a.ko.Spec.Tags), ToACKTags(b.ko.Spec.Tags)) {
 		delta.Add("Spec.Tags", a.ko.Spec.Tags, b.ko.Spec.Tags)
 	}
-	if !reflect.DeepEqual(a.ko.Spec.Taints, b.ko.Spec.Taints) {
+	
+	if a.ko.Spec.Taints != nil && a.ko.Spec.Taints == nil || a.ko.Spec.Taints == nil && a.ko.Spec.Taints != nil {
 		delta.Add("Spec.Taints", a.ko.Spec.Taints, b.ko.Spec.Taints)
+	} else if a.ko.Spec.Taints != nil {
+		if len(a.ko.Spec.Taints) != len(b.ko.Spec.Taints) {
+			delta.Add("Spec.Taints", a.ko.Spec.Taints, b.ko.Spec.Taints)
+		} else {
+			for _, taintA := range a.ko.Spec.Taints {
+				var matched = false
+				for _, taintB := range b.ko.Spec.Taints {
+					if reflect.DeepEqual(taintA, taintB) {
+						matched = true
+						break
+					}
+				}
+				if !matched {
+					delta.Add("Spec.Taints", a.ko.Spec.Taints, b.ko.Spec.Taints)
+					break
+				}
+			} 
+		}
 	}
+
```

Comment:
This code is generated, so next time this controller is regenerated we will lose all this code :). Ideally you need to instruct the code generator on how to inject the new logic into `delta.go`
1. First we need to drop the current implementation: To do that you'll have add `compare: false` into the `NodeGroup` resource:
```yaml
  Nodegroup:
    fields:
      compare: false
```

2. We need to ask the code-generator to inject a new comparison function that will use your implementation to find the delta (which is already done - so no need to worry about this)
3. Add your logic to https://github.com/aws-controllers-k8s/eks-controller/blob/main/pkg/resource/nodegroup/hook.go#L80 (note that this function is already doing some defaulting)

Make sure to hit `make build-controller SERVICE=eks` from the code-generator repository everytime you modify the `generator.yaml`